ToolBox/SamcoDataDownloader/SamcoDataDownloader.cs

```diff
@@ -0,0 +1,101 @@
+ * limitations under the License.
+*/
+
+using QuantConnect.Brokerages.Samco;
+using QuantConnect.Configuration;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Logging;
+using QuantConnect.Securities;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.ToolBox.SamcoDataDownloader
+{
+    public class SamcoDataDownloaderProgram
+    {
+        /// <summary>
+        /// Samco Data Downloader Toolbox Project For LEAN Algorithmic Trading Engine. By Balamurali
+        /// Pandranki a.k.a @itsbalamurali
+        /// </summary>
+        public static void SamcoDataDownloader(IList<string> tickers, string market, string resolution, string securityType, DateTime startDate, DateTime endDate)
+        {
+            if (resolution.IsNullOrEmpty() || tickers.IsNullOrEmpty())
+            {
+                Log.Error("SamcoDataDownloader ERROR: '--tickers=', --securityType, '--market' or '--resolution=' parameter is missing");
+                Log.Error("--tickers=eg JSWSTEEL,TCS,INFY");
+                Log.Error("--market=MCX/NSE/NFO/CDS/BSE");
+                Log.Error("--security-type=Equity/Future/Option");
+                Log.Error("--resolution=Minute/Hour/Daily/Tick");
+                Environment.Exit(1);
+            }
+            try
+            {
+                var _samcoAPI = new SamcoBrokerageAPI();
+                _samcoAPI.Authorize(Config.Get("samco-client-id"), Config.Get("samco-client-password"), Config.Get("samco-year-of-birth"));
+
+                var castResolution = (Resolution)Enum.Parse(typeof(Resolution), resolution);
+                var castSecurityType = (SecurityType)Enum.Parse(typeof(SecurityType), securityType);
+
+                if (castSecurityType == SecurityType.Forex || castSecurityType == SecurityType.Cfd || castSecurityType == SecurityType.Crypto || castSecurityType == SecurityType.Base)
+                {
+                    throw new ArgumentException("Invalid security type: " + castSecurityType);
+                }
+
+                if (startDate >= endDate)
+                {
+                    throw new ArgumentException("Invalid date range specified");
+                }
+
+                // Load settings from config.json and create downloader
+                var dataDirectory = Globals.DataFolder;
+                var marketHoursDatabase = MarketHoursDatabase.FromDataFolder();
+                var symbolMapper = new SamcoSymbolMapper();
+
+                foreach (var pair in tickers)
+                {
+                    // Download data
+                    var pairObject = Symbol.Create(pair, castSecurityType, market);
+                    var exchange = symbolMapper.GetDefaultExchange(pairObject);
+                    var scrip = symbolMapper.SamcoSymbols.Where(x => x.Name.ToUpperInvariant() == pairObject.ID.Symbol).First();
+                    var isIndex = scrip.Instrument == "INDEX";
```

Comment:
if `isIndex` is true, should it be a Index security type from Leans perspective?